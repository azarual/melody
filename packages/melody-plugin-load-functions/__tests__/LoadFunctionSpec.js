import { compile, toString } from 'melody-compiler';
import { extension as coreExtension } from 'melody-extension-core';
import idomPlugin from 'melody-plugin-idom';
import skipIfPlugin from '../src';

test('load should compile to import()', () => {
    expect(render(`{{ load('fancy-component') }}`)).toMatchSnapshot();
});

test('loadSync should compile to import statement', () => {
    expect(render(`{{ loadSync('fancy-component') }}`)).toMatchSnapshot();
});

function render(template) {
    try {
        const jsTemplate = compile(
            'test.twig',
            template,
            coreExtension,
            idomPlugin,
            skipIfPlugin
        );
        return toString(jsTemplate, template).code;
    } catch (e) {
        return e.message;
    }
}
