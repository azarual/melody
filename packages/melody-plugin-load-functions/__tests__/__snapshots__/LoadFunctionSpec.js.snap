// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load should compile to import() 1`] = `
"export const _template = {};

_template.render = function (_context) {
  import(\\"fancy-component\\");
};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

export default function Test(props) {
  return _template.render(props);
}"
`;

exports[`loadSync should compile to import statement 1`] = `
"import \\"fancy-component\\";
export const _template = {};

_template.render = function (_context) {};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

export default function Test(props) {
  return _template.render(props);
}"
`;
